test/powershell/Modules/Microsoft.PowerShell.Core/History.Tests.ps1

```diff
@@ -14,22 +14,33 @@ Describe "History cmdlet test cases" -Tags "CI" {
             $ps.commands.clear()
             $null = $ps.addcommand("Get-Location").Invoke($null, $setting)
             $ps.commands.clear()
+            $null = $ps.addscript("2+2").Invoke($null, $setting)
+            $ps.commands.clear()
         }
         AfterEach {
             $ps.Dispose()
         }
         It "Get-History returns proper history" {
             # for this case, we'll *not* add to history
             $result = $ps.AddCommand("Get-History").Invoke()
-            $result.Count | Should -Be 3
+            $result.Count | Should -Be 4
             $result[0].CommandLine | Should -BeExactly "Get-Date"
             $result[1].CommandLine | Should -Be "1+1"
             $result[2].CommandLine | Should -BeExactly "Get-Location"
+            $result[3].CommandLine | Should -Be "2+2"
         }
         It "Invoke-History invokes proper command" {
             $result = $ps.AddScript("Invoke-History 2").Invoke()
             $result | Should -Be 2
         }
+        It "Invoke-History executes multiple IDs" {
+            $result = $ps.AddScript("Invoke-History 2,4").Invoke()
+            $result | Should -Be @(2, 4)
+        }
+        It "Invoke-History invokes the last command by default" {
+            $result = $ps.AddScript("Invoke-History").Invoke()
+            $result | Should -Be 4
+        }
```

Comment:
Am I misremembering or does Invoke-History also accept piped IDs? Probably best to have that included in at least once test if so.

Reply:
I don't think it does. This is the kind of experience you get.

```powershell
PS /workspaces/PowerShell> clear-history
PS /workspaces/PowerShell> write-output "hi"        
hi
PS /workspaces/PowerShell> write-output "lo"        
lo
PS /workspaces/PowerShell> get-history

  Id     Duration CommandLine
  --     -------- -----------
  21        0.001 clear-history
  22        0.001 write-output "hi"
  23        0.001 write-output "lo"

PS /workspaces/PowerShell> @(22,23) | invoke-history
Invoke-History: The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its properties do not match any of the parameters that take pipeline input.
Invoke-History: The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its properties do not match any of the parameters that take pipeline input.
get-history

  Id     Duration CommandLine
  --     -------- -----------
  21        0.001 clear-history
  22        0.001 write-output "hi"
  23        0.001 write-output "lo"
  24        0.003 get-history

PS /workspaces/PowerShell> 
```